src/server/src/main/java/io/cassandrareaper/service/RepairScheduleService.java

```diff
@@ -94,6 +100,71 @@ public RepairSchedule storeNewRepairSchedule(
         .segmentCountPerNode(segmentCountPerNode)
         .owner(owner);
 
-    return context.storage.addRepairSchedule(scheduleBuilder);
+    RepairSchedule repairSchedule = context.storage.addRepairSchedule(scheduleBuilder);
+    registerScheduleMetrics(repairSchedule.getId());
+    return repairSchedule;
+  }
+
+  public void deleteRepairSchedule(UUID repairScheduleId) {
+    unregisterScheduleMetrics(repairScheduleId);
+    context.storage.deleteRepairSchedule(repairScheduleId);
+  }
+
+  private void registerRepairScheduleMetrics(Collection<RepairSchedule> allRepairSchedules) {
+    allRepairSchedules.forEach(schedule -> registerScheduleMetrics(schedule.getId()));
+  }
+
+
+  private void registerScheduleMetrics(UUID repairScheduleId) {
+    RepairSchedule schedule = context.storage.getRepairSchedule(repairScheduleId).get();
+    RepairUnit repairUnit = context.storage.getRepairUnit(schedule.getRepairUnitId());
+    String metricName = metricName("millisSinceLastRepairForSchedule",
+            repairUnit.getClusterName(),
+            repairUnit.getKeyspaceName(),
+            schedule.getId());
+
+    if (!context.metricRegistry.getMetrics().containsKey(metricName)) {
+      context.metricRegistry.register(metricName, getMillisSinceLastRepairForSchedule(schedule.getId()));
+    }
+  }
+
+  private void unregisterScheduleMetrics(UUID repairScheduleId) {
+    RepairSchedule schedule = context.storage.getRepairSchedule(repairScheduleId).get();
+    RepairUnit repairUnit = context.storage.getRepairUnit(schedule.getRepairUnitId());
+    String metricName = metricName("millisSinceLastRepairForSchedule",
+        repairUnit.getClusterName(),
+        repairUnit.getKeyspaceName(),
+        schedule.getId());
+
+    if (context.metricRegistry.getMetrics().containsKey(metricName)) {
+      context.metricRegistry.remove(metricName);
+    }
+  }
+
+  private Gauge<Long> getMillisSinceLastRepairForSchedule(UUID repairSchedule) {
+    return () -> {
+      Optional<RepairSchedule> schedule = context.storage.getRepairSchedule(repairSchedule);
+
+      Long lastRepairDate = schedule.orElseThrow(() -> new IllegalArgumentException("Repair schedule not found"))
+          .getRunHistory().stream()
+          .map(uuid -> context.storage.getRepairRun(uuid))
+          .filter(Optional::isPresent)
+          .map(Optional::get)
+          .map(RepairRun::getEndTime)
+          .filter(Objects::nonNull)
+          .max(DateTime::compareTo)
+          .map(dateTime -> DateTime.now().getMillis() - dateTime.getMillis())
```

Comment:
You are right - I used the same reasoning:

When a schedule has never been successfully repaired, no data about a repair is present, so what value should the metric have? 
The value 0 would indicate that it was last repaired 0 milliseconds ago, which is not the case. 
The value epoch, would indicate that a repair happened at the beginning of time, which is also not the case, but fits alert rules better.

The (Prometheus) alert expression I use looks like this:
```
io_cassandrareaper_service_RepairScheduleService_millisSinceLastRepairForSchedule > 1000 * 60 * 60 * 24 * 2
```
It triggers if there has not been a repair in the last 2 days. If there has never been a repair, it would trigger as well.